runtime/src/accounts.rs

```diff
@@ -230,9 +230,9 @@ impl Accounts {
                     return Err(TransactionError::ProgramAccountNotFound);
                 }
             };
-            if !program.executable || program.owner == Pubkey::default() {
```

Comment:
Why get of the enforcement of program owner?

Reply:
Because it will loop again and return `ProgramAccountNotFound` as tested here: https://github.com/solana-labs/solana/pull/9047/files#diff-0104dfa96426a3a49428815872a0356eR1200